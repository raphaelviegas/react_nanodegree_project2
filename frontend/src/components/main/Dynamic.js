import React from 'react'
import PropTypes from 'prop-types'
import SwitchRoutes from './SwitchRoutes'

const Dynamic = (props) => {
  return (
    <div>
      {props.loading === true
        ? <div className="progress"><div className="indeterminate"></div></div>
        : <div className="section"><SwitchRoutes /></div>
      }
    </div>
  )
}

Dynamic.propTypes = {
  loading: PropTypes.bool.isRequired
}

export default Dynamic