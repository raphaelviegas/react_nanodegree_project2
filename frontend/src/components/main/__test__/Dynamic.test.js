import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import Dynamic from '../Dynamic'

describe('<Dynamic />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<Dynamic loading={true} />)
    })

    it('Should render a progress bar when the posts are loading (props.loading === true)', () => {
        const loadingBar = <div className="progress"><div className="indeterminate"></div></div>
        expect(wrapper.containsMatchingElement(loadingBar)).toEqual(true)
    })

    it('SNAPSHOT - Should render a progress bar when the posts are loading (props.loading === true)', () => {
        const tree = renderer.create(
            <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
                <Dynamic loading={true} />
            </MemoryRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Should render a  <SwitchRoutes> when props.loading === false', () => {
        wrapper.setProps({ loading: false })
        expect(wrapper.find('SwitchRoutes').length).toEqual(1)
    })

})
