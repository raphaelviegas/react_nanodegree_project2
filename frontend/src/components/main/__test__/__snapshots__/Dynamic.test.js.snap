// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dynamic /> SNAPSHOT - Should render a progress bar when the posts are loading (props.loading === true) 1`] = `
<div>
  <div
    className="progress"
  >
    <div
      className="indeterminate"
    />
  </div>
</div>
`;
